ovsdb/client.go

```diff
@@ -82,10 +104,26 @@ func New(conn net.Conn, options ...OptionFunc) (*Client, error) {
 	// Set up callbacks.
 	client.callbacks = make(map[int]callback)
 
-	// Start up any background routines.
+	// Start up any background routines, and enable canceling them via context.
+	ctx, cancel := context.WithCancel(context.Background())
+	client.cancel = cancel
+
 	var wg sync.WaitGroup
 	wg.Add(1)
```

Comment:
Nit: I feel like it flows a little better to have this `wg.Add(1)` one line above its associated `go func()` (for `client.Listen()` below).

Reply:
Yeah, so the problem is that I want to make sure that the waitgroup has its number of goroutines to track already set up before either of them actually fire.  It feels race condition prone otherwise.

Reply:
Ah yeah, I see what you mean -- it's possible that something could call `Wait` in the middle of all this. I guess I was thinking that wasn't possible, since this is happening in `New()`, but no, your reasoning still makes sense.